src/main/scala/com/databricks/spark/xml/util/XmlFile.scala

```diff
@@ -93,7 +93,12 @@ private[xml] object XmlFile {
         override def next: String = {
           if (iter.nonEmpty) {
             if (firstRow) {
-              indentingXmlWriter.writeStartElement(options.rootTag)
+              if (options.header != null) {
+                writer.append(options.header)
+              }
+              if (!options.rootTag.isEmpty) {
```

Comment:
I don't think we can escape writing the root tag. It would not be valid XML otherwise.

Reply:
Depends on schema. Root tag is required when exporting a simple table. In case of a complex schema e.g.:
```
StructType(
  StructField("person",
    ArrayType(
      StructType(...)
    )),
  StructField("animal",
    ArrayType(
      StructType(...)
    ))
)
```
where the table has a single row with array columns the produced xml has an extra root tag:
```
<root>
  <row>
    <person> ... </person>
    <person> ... </person>
    <animal> ... </animal>
    <animal> ... </animal>
  </row>
<root>
```
This is definitely a useful feature.

Reply:
I don't think so. You would have a doc like:

```
  <row>
    <person> ... </person>
    <person> ... </person>
    <animal> ... </animal>
    <animal> ... </animal>
  </row>
  <row>
    <person> ... </person>
    <person> ... </person>
    <animal> ... </animal>
    <animal> ... </animal>
  </row>
  ...
```
There would be no root element in the XML file.

Reply:
My mistake, it doesn't depend on schema complexity but on number of rows.

Table with only one row produces a single row element therefore it can act as the root element. I know that my table has a single row because it's constructed that way: multiple tables are joined together and collapsed into arrays of structs preserving their hierarchy.

Maybe there should be some check throwing an exception if the source table has multiple rows.

Reply:
OK, I don't really object. It won't be the default and hey if someone really wants to do it, I guess they can if they know what they're doing.